src/api/data_types/chunking/dif.rs

```diff
@@ -25,5 +27,66 @@ pub struct ChunkedDifResponse {
     pub dif: Option<DebugInfoFile>,
 }
 
-pub type AssembleDifsRequest<'a> = HashMap<Digest, ChunkedDifRequest<'a>>;
+pub struct AssembleDifsRequest<'a>(HashMap<Digest, ChunkedDifRequest<'a>>);
 pub type AssembleDifsResponse = HashMap<Digest, ChunkedDifResponse>;
+
+impl<'a> ChunkedDifRequest<'a> {
+    /// Create a new ChunkedDifRequest with the given name, chunk hashes,
+    /// and total hash for the entire file.
+    pub fn new(name: &'a str, chunks: &'a [Digest], hash: Digest) -> Self {
+        Self {
+            name,
+            chunks,
+            hash,
+            debug_id: None,
+        }
+    }
+
+    /// Set the provided debug_id on the request, or clear it if
+    /// `None` is passed.
+    pub fn with_debug_id(mut self, debug_id: Option<DebugId>) -> Self {
+        self.debug_id = debug_id;
+        self
+    }
+}
+
+impl AssembleDifsRequest<'_> {
+    /// Strips the debug_id from all requests in the request. We need
+    /// to strip the debug_ids whenever the server does not support chunked
+    /// uploading of PDBs, to maintain backwards compatibility. The
+    /// calling code is responsible for calling this function when needed.
+    ///
+    /// See: https://github.com/getsentry/sentry-cli/issues/980
+    /// See: https://github.com/getsentry/sentry-cli/issues/1056
+    pub fn strip_debug_ids(&mut self) {
+        for r in self.0.values_mut() {
+            r.debug_id = None;
+        }
+    }
+}
+
+impl<'a, T> FromIterator<T> for AssembleDifsRequest<'a>
+where
+    T: Into<ChunkedDifRequest<'a>>,
+{
+    fn from_iter<I>(iter: I) -> Self
+    where
+        I: IntoIterator<Item = T>,
+    {
+        Self(
+            iter.into_iter()
+                .map(|obj| obj.into())
+                .map(|r| (r.hash, r))
+                .collect(),
+        )
+    }
+}
+
+impl Serialize for AssembleDifsRequest<'_> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        self.0.serialize(serializer)
+    }
+}
```

Comment:
Why is this manually implemented? Isn't this the same as what would be derived?

Reply:
I thought it might be different, because I want serde just to serialize the inner `HashMap` instead of wrapping it in another layer. But apparently, I can use `#[serde(transparent)]` to guarantee that the derived serialization simply serializes the inner `HashMap`, so I will do that and remove the manual implementation.

Reply:
That's also the default behavior of `serde`.

Reply:
> That's also the default behavior of serde.

From my very brief research, that is indeed the case for `serde_json`, and it is encouraged for all serde data formats to behave this way, but it is not a guarantee that all serializers serialize Newtype structs as the type they are wrapping. 

From [Serde's docs](https://serde.rs/impl-serialize.html#serializing-a-struct):

> Data formats are encouraged to treat newtype structs as insignificant wrappers around the inner value, serializing just the inner value.

To guarantee that we serialize just the wrapped value, we need to use `#[serde(transparent)]`.

Reply:
That's fair, and smart of you to check.